### PR TITLE
fix save menu displayed in snapshot if save while demo is playing

### PR DESCRIPTION
**Pull request by rfomin**
Fix this issue:
woof0002

### CODE REVIEW

File: src/g_game.c [hunk 1]
```diff
@@ -2186,7 +2186,7 @@ void G_Ticker(void)
   if (demoplayback && sendsave)
     {
       sendsave = false;
-      G_DoSaveGame();
+      gameaction = ga_savegame;
```
Comment (fabiangreffrath) [inline src/g_game.c:2189]: This will postpone game saving by one tic. Is this considered safe?

Comment (rfomin) [reply src/g_game.c:2189]: I think so, but I'm not sure. This is how a normal non-demo save works.

**Comment by fabiangreffrath** [comment]
Maybe we should move the creation of the snapshot into `D_Display()` right before the `M_Drawer()` call if `gameaction == ga_savegame`. Do you think this could work?

**Comment by rfomin** [comment]
> Maybe we should move the creation of the snapshot into `D_Display()` right before the `M_Drawer()` call if `gameaction == ga_savegame`. Do you think this could work?

I think I could. But the non-demo save is also delayed by one tick, why is there an exception for the demo save? Maybe there is no difference.

**Comment by fabiangreffrath** [comment]
> I think I could. But the non-demo save is also delayed by one tick, why is there an exception for the demo save? Maybe there is no difference.

Agreed, let's go that way.

**Pull request merged**